src/Aquifer.Data/Entities/ResourceContentVersionSimilarityScore.cs

```diff
@@ -0,0 +1,22 @@
+namespace Aquifer.Data.Entities;
+
+public class ResourceContentVersionSimilarityScore
+{
+    public int Id { get; set; }
+    public int BaseVersionId { get; set; }
+    public ResourceContentVersionTypes BaseVersionType { get; set; }
+    public int ComparedVersionId { get; set; }
+    public ResourceContentVersionTypes ComparedVersionType { get; set; }
+    public ResourceContentStatus Status { get; set; }
+    public double SimilarityScore { get; set; }
+    
+    [SqlDefaultValue("getutcdate()")]
+    public DateTime Created { get; set; } = DateTime.UtcNow;
+}
+
+public enum ResourceContentVersionTypes
+{
+    Base,
```

Comment:
We can compare ant two `ResourceContentVersions` , `...MachineTranslations` and `...Snapshots` and have to be able to identify which type of resource we are comparing. Since it can be any of these three, we cannot have a foreign key here.

Reply:
I like to have `None` as 0, since the default value for an enum is 0. That way it doesn't take on a value if someone doesn't set it.

Reply:
This should never have 'none'. It has to be either a base `ResourceContentVersion`, a `ResourceContentVersionMachineTranslation`, or `ResourceContentVersionSnapshot`. Otherwise, the comparison cannot happen.

Reply:
That's my point though. If someone doesn't set it it has a default value that has meaning. So the default shouldn't have meaning.

Reply:
Ah, bc Mssql doesn't have an enum type. It makes it an int...and that could cause what you described.